site/app/controllers/course/CourseMaterialsController.php

```diff
@@ -129,13 +133,14 @@ public function downloadCourseMaterialZip($dir_name, $path) {
             $this->core->getOutput()->showError("You do not have access to this folder");
             return false;
         }
-        $zip->close();
+        $zip_stream->finish();
         header("Content-type: application/zip");
         header("Content-Disposition: attachment; filename=$zip_file_name");
         header("Content-length: " . filesize($zip_name));
         header("Pragma: no-cache");
         header("Expires: 0");
         readfile("$zip_name");
+        unlink($zip_name);
```

Comment:
Is this already handled by `register_shutdown_function('unlink', $zip_name);` on line 92?

Reply:
I realized lines 137-143 are unnecessary when using zip stream and were causing problems with some downloads, they have been removed